Сделал навигацию с адаптивом

Rubsee opened the pull request.

Rubsee committed:
Сделал навигацию с адаптивом

captain-zsa commented on source/sass/blocks/main-nav.scss:
```diff
@@ -0,0 +1,129 @@
+            // margin-left: 28px;
+        }
+
+        @media (min-width: 768px) {
```
Зачем так делать? Напишите этот media прямо внутри img предыдущего.

captain-zsa commented on source/sass/blocks/main-nav.scss:
```diff
@@ -0,0 +1,129 @@
+            height: 16px;
+        }
+
+        @media (min-width: 768px) {
```
Та же история.

captain-zsa commented on source/sass/blocks/main-nav.scss:
```diff
@@ -0,0 +1,129 @@
+            width: 74px;
+            height: 25px;
+
+            @media (min-width: 768px) {
```
Вот точно также и надо сделать в предыдущих местах с img и media

captain-zsa commented on source/sass/blocks/main-nav.scss:
```diff
@@ -0,0 +1,129 @@
+        display: none;
+
+        @media (min-width: 768px) {
+            @include clear-list;
```
Я бы вынес в самое начало перед display: none;

captain-zsa commented on source/index.html:
```diff
@@ -1,13 +1,46 @@
+</head>
+
+<body>
+  <header class="header">
```
Не очень ясно зачем вам header, если дальше всё внутри nav.

Логичнее и лучше сделать так – оставить header, а nav сделать только вокруг ul.main-nav__list потому что основная навигация всё же это именно список.

Rubsee committed:
Сделал работу над ошибками в ветке task2

1. Переместил @media в img
- у logo
- и кнопок которые в buttons
2. В list'е опустил display:none ниже @media
3. Оставил header и обернул list в nav

Rubsee committed:
Сделал правки

Сделал правки после созвона
- поправил классы для header'a и main-nav
- вынес кнопку меню(button__burger) и спозиционировал абсолютно, относительно header контейнера

captain-zsa commented on source/sass/style.scss:
```diff
@@ -4,4 +4,7 @@
 @import 'blocks/container.scss';
-@import 'blocks/main-nav.scss';
\ No newline at end of file
+@import 'blocks/header.scss';
```
Чем более объемный блок – тем ниже он должен подключаться. К примеру внутри header у вас есть main-nav. Значит main-nav должен подключаться выше header. Почему так – потому что если у вас будет элемент
вот с таким набором классов
`class="header__nav main-nav"`

И к примеру будет
.header__nav {
  margin: 10px;
}

.main-nav {margin: 20px;}

вы очевидно ожидаете, что для main-nav стили менее приоритетны (так как именно header__nav должен заниматься регулированием того, как main-nav себя внутри header ведет). НО проблема в том, что при вашем варианте подключения это всё сработает не так. Именно стили от main-nav будут более приоритетны, потому что они подключаются ниже, чем стили header. А то, что ниже – то приоритетней.

Надеюсь, спич мой понятен. Если нет – в чате спросите, я запишу короткий ролик, покажу в чём проблема.

Естественно, это же относится и к другим файлам, которые вы подключаете. Принцип прост – чем меньше будет внутри бэм-блока других бэм-блоков – тем выше он должен быть подключен. А бэм-блоки, внутри которых нет других бэм-блоков, должны быть подключены вообще выше всего.

К примеру если сделать общий бэм-блок .button (у нас вон в макете есть черные кнопочки и их разновидности – красные, белые и синие). Так вот, внутри кнопки ну никаких других блоков априори быть не может. Внутри кнопки только текст и всё на этом. Значит файл button.scss должен быть подключен как можно выше. А вот к примеру внутри header у нас теоретически может появиться button и другие блоки. Значит мы его подключаем ниже.

Rubsee committed:
Поправил порядок подключения
